### DISCUSSION TITLE
使用createSchemaField创建的SchemaField实例，schema props更新，组件不重新更新

### DISCUSSION THREAD

**Discussion by tywinMa** [alibaba/formily#4063] [Q&A 2.x]
如图我把key为input的组件换为select（或者修改其他属性）的时候，组件并不会自动更新，只有我修改整个properties才会有相应，这是什么原因呢
![image](https://github.com/alibaba/formily/assets/104997217/cc649e3a-39f3-446e-97fd-c27be3e77395)

https://stackblitz.com/edit/stackblitz-starters-6nfjps?file=src%2FApp.tsx

**Comment by yanhao98** [comment]
https://core.formilyjs.org/zh-CN/api/models/field#setcomponent
